Improve the yt-dlp.sh script

fstirlitz opened the pull request.

pukkandan commented:
> No description provided

. . .

Please explain the intention behind the proposed changes
And don't ignore the template in the future

fstirlitz commented:
Everything is in the commit message; it’s a pretty trivial change. The most important bit (ahem) is the execute permission on the script, so that it can be executed directly. Along the way, I added some quoting, changed the interpreter to plain `sh` and added `exec` to chainload the Python interpeter instead of spawning it as a subprocess of the shell. I hope I don’t need to explain how shell scripts work.

pukkandan commented:
oh, sorry. I didn't think to read the commit description.
